internal/server/scriptlet/qemu.go

```diff
@@ -15,10 +16,52 @@ import (
 	"github.com/lxc/incus/v6/shared/logger"
 )
 
+// cfgSection is a temporary struct to hold QEMU configuration sections with Entries of type
+// map[string]string instead of []cfg.Entry. This type should be moved to the cfg package in the
+// future.
+type cfgSection struct {
+	Name    string            `json:"name"`
+	Comment string            `json:"comment"`
+	Entries map[string]string `json:"entries"`
+}
+
+// marshalConf marshals a configuration into a []map[string]any.
+func marshalConf(conf any) ([]map[string]any, error) {
+	jsonConf, err := json.Marshal(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	var newConf []map[string]any
+	err = json.Unmarshal(jsonConf, &newConf)
+	if err != nil {
+		return nil, err
+	}
+
+	return newConf, nil
+}
```

Comment:
Since this isn't in a qemu-specific package (it's in `scriptlet`), the type and function should probably have a `qemu` prefix so it's clear that it's related to QEMU.

Reply:
Yeah you’re right. I could even put the whole scriptlet in its own subpackage. We’ve not really been consistent on this, as the “subpackagification” has been the solution to break dependency cycles. Let’s keep it this way and prefix everything with `qemu` for now.